### PR TITLE
feat(Statistics): Hide Toolbar on Scroll

### LINKED ISSUE #14991: Hide the statistics toolbar on scroll
Currently, the graphs don't fit in the screen when in the landscape mode.

https://forums.ankiweb.net/t/suggestion-to-redesign-the-statistics-tab/37129/6

Dae said:
> It fits on an iPhone. You could ask the AnkiDroid devs to decrease the height of the top area and/or hide the status bar.

### PR DESCRIPTION
**Pull request by neeldoshii**
## Fixes
* Fixes #14991 

* Because of old stale PR created this new PR for historic reference of old refer PR #15068 

## Testing Instructions
1. Open the Hamburger Menu
2. Open the Statistic tab
3. Perform Vertical Scroll and check the toolbar visiblity.

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/pages/Statistics.kt [hunk 1]
```diff
@@ -34,10 +36,18 @@ class Statistics : PageFragment() {
...
+        savedInstanceState: Bundle?
+    ): View? {
+        return inflater.inflate(R.layout.statistics, container, false)
+    }
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
-        view.findViewById<MaterialToolbar>(R.id.toolbar)?.apply {
+        val toolbar = view.findViewById(R.id.toolbar) as MaterialToolbar
+        toolbar.apply {
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/pages/Statistics.kt:50]: There's a weak IDE warning here, in future see if these can be resolved (as they make the code easier to maintain)

```suggestion
        view.findViewById<MaterialToolbar>(R.id.toolbar).apply {
```

**Review by david-allison — APPROVED** [review]
Thanks!

**Comment by david-allison** [comment]
Thanks!

**Comment by neeldoshii** [comment]
@ShridharGoel Here is the video, I found one bug and fixed it.

https://github.com/ankidroid/Anki-Android/assets/60827173/87988206-7989-468f-b9b8-80356f863122

File: AnkiDroid/src/main/res/layout/statistics.xml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.coordinatorlayout.widget.CoordinatorLayout
```
Comment (BrayanDSO) [inline AnkiDroid/src/main/res/layout/statistics.xml:2]: Could you try adding a fitsSystemWindows=true into the CoordinatorLayout so the status bar match the toolbar?

**Review by BrayanDSO — CHANGES_REQUESTED** [review]

File: AnkiDroid/src/main/res/layout/statistics.xml (hunk 2 above)
Comment (neeldoshii) [reply AnkiDroid/src/main/res/layout/statistics.xml:2]: I did add it previously, but the slight preview was getting shown on status bar. So in next commit I removed it.

Comment (BrayanDSO) [reply AnkiDroid/src/main/res/layout/statistics.xml:2]: Wdym by slight preview?

Comment (neeldoshii) [reply AnkiDroid/src/main/res/layout/statistics.xml:2]: Check this.

https://github.com/ankidroid/Anki-Android/assets/60827173/a4955a30-1002-4efa-8aa9-0bf7c7ab2be4

```
<androidx.coordinatorlayout.widget.CoordinatorLayout
    xmlns:android="http://schemas.android.com/apk/res/android"
    xmlns:app="http://schemas.android.com/apk/res-auto"
    android:layout_width="match_parent"
    android:layout_height="match_parent"
    android:fitsSystemWindows="true">
 ```

Comment (BrayanDSO) [reply AnkiDroid/src/main/res/layout/statistics.xml:2]: Okay. Keep it removed, then add an issue to do the status bar match

**Comment by BrayanDSO** [comment]
Do you know how to `git rebase` and squash your commits? If not, the internet have infinite tutorials about it. 

Squashing all the commits would be a good exercise for you

**Review by BrayanDSO — APPROVED** [review]

**Comment by david-allison** [comment]
Waiting for @neeldoshii to squash + force push

**Comment by neeldoshii** [comment]
Working on it now.

**Comment by neeldoshii** [comment]
Done @david-allison

**Comment by david-allison** [comment]
Cheers! Makes it much easier to review and merge

**Pull request merged**